frontend/app/models/legislator.js

```diff
@@ -37,7 +46,95 @@ var Legislator = DS.Model.extend({
 
   formattedChamber: function(){
     return this.get('chamber').capitalize();
-  }.property('chamber')
+  }.property('chamber'),
+
+  formattedParty: function(){
+    var partyChar = this.get('party');
+    switch (partyChar){
+      case 'R':
+        return 'Republican';
+      case 'D':
+        return 'Democrat';
+      case 'I':
+        return 'Independent';
+      default:
+        return 'Other';
+    };
+  }.property('party'),
+
+  termLength: function(){
+    var termStart = this.get('term_start'),
+        dateArray = termStart.split('-'),
+        string;
+        
+    string = getDateString( dateArray[1], dateArray[0] );
+
+    return string;
+  }.property('term_start'),
+
+  termLeft: function(){
+    var termEnd = this.get('term_end'),
+        dateArray = termEnd.split('-'),
+        string;
+
+    string = getDateString( dateArray[1], dateArray[0] );
+        
+    return string;    
+  }.property('term_end'),
+
+  twitterLink: function(){
+    var twitterID = this.get('twitter_id'),
+        fullLink;
+
+    fullLink = "http://www.twitter.com/" + twitterID;
+
+    return fullLink;
+  }.property('twitter_id'),
+  
+  youtubeLink: function(){
+    var youtubeID = this.get('youtube_id'),
+        fullLink;
+
+    fullLink = "http://www.youtube.com/user/"+youtubeID
+
+    return fullLink;
+  }.property('youtube_id'),
+
+   emailLink: function(){
+    var email = this.get('oc_email'),
+        fullLink;
+
+    fullLink = "mailto:"+email
+
+    return fullLink;
+  }.property('oc_email')
 });
 
+var getDateString = function(month, year) {
```

Comment:
Hmm I'm not sure. Take a look at cli's docs for where to put helper files. But I think we can import from any file using the ES6 require stuff. If so we should prolly make a helpers folder. —
Sent from Mailbox

On Mon, Jan 19, 2015 at 8:43 PM, Stockbridge <notifications@github.com>
wrote:

>>  });
>>  
>> +var getDateString = function(month, year) {
> which file should I put them in?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jimbol/on-behalf/pull/1/files#r23200140